dts/bindings/gnss/gnss-nmea-generic.yaml

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Generic GNSS NMEA receiver
+
+  Implement a generic NMEA based GNSS device.
+
+  Example configuration:
+
+  &uart0 {
+          current-speed = <9600>;
+          ...
+          gnss: gnss-nmea-generic {
+                  compatible = "gnss-nmea-generic";
+          };
+  };
+
+compatible: "gnss-nmea-generic"
+
+include:
+  - uart-device.yaml
+
+properties:
+  nmea-timeout-ms:
+    type: int
+    default: 500
+    description: |
+      Synchronization timeout for NMEA sentences. The NMEA parser is expecting
+      to receive a GGA and RMC sentences within this time frame to publish a
+      location data. Set accordingly to the UART datarate and location
+      reporting frequency. Defaults to 200ms if unspecified.
```

Comment:
500ms :)

Reply:
Fixed. Do you think it would be possible to have the driver measure few messages and tweak this automatically?

Reply:
I think so, if we track the max delta between messages we are listening to, then we could set the timeout to half that delta.
```
0.10s RMC
0.15s GGA +0.05s
0.35s GSV +0.20s
0.36s GSV +0.01s
0.37s GSV +0.01s
0.38s GSV +0.01s
0.39s GSV +0.01s
1.00s RMC +0.61s
```
Max delta / 2 = 305ms
```
0.10s RMC
0.15s GGA +0.05s
0.35s GSV +0.20s
0.36s GSV +0.01s
0.37s GSV +0.01s
0.38s GSV +0.01s
0.39s GSV +0.01s
10.00s RMC +9.61s
```
Max delta / 2 = 4805ms

We can use RMC to recalculate the automatic delta calculation, to handle the user changing the fix rate. The first 1 to 2 data points would be discarded since the delta is not known

Reply:
Now that I'm looking at it, it can just be fix rate / 2 :D

Reply:
but then if the link is slow enough (guess 4800 is still common on these) then a transmission I think could use well over 50% of the uart bandwidth and get close to the 1s fix. Maybe it could use like 90% fix rate and then reset once the message is published?

Reply:
The issue is the timing, we don't know which message comes first, RMC, GGA or GSV, if we don't get the timing right, we could be publishing an RMC message which contains data which does not reflect GGA or GSV. The timeout is for re-synchronization, so we know when to publish

Reply:
If we knew which came first, we could forego the timeout completely

Reply:
Ok but then it sounds like the timeout option would only be usable on fast channels, if more than 50% of the uart time is utilized you'd end up never timing out and always reporting one of the two out of sync I imagine. But then the only solution would be to manually specify which one comes first... wondering how gps handles that.